### ISSUE TITLE
Unable to use eslint-disable-next-line for valid-jsdoc

### ISSUE THREAD

**Issue by Krinkle** [eslint/eslint#10095]
* **ESLint version:** 4.9.0
* **Node.js version:** 9.8.0
* **npm version:** 5.6.0

**Please show your full configuration:**

<details>
<summary>Configuration (eslint-config-wikimedia@v0.5.0)</summary>
```json
{
  "extends": "eslint:recommended",
  "env": {},
  "globals": {},
  "rules": {
    "array-bracket-spacing": [ "error", "always" ],
    "block-spacing": "error",
    "brace-style": [ "error", "1tbs", { "allowSingleLine": true } ],
    "camelcase": [ "error", { "properties": "always" } ],
    "comma-dangle": [ "error", "never" ],
    "comma-spacing": [ "error", { "before": false, "after": true } ],
    "comma-style": [ "error", "last" ],
    "computed-property-spacing": [ "error", "always" ],
    "curly": [ "error", "all" ],
    "dot-location": [ "error", "property" ],
    "dot-notation": [ "error" ],
    "eol-last": "error",
    "eqeqeq": "error",
    "func-call-spacing": "error",
    "indent": [ "error", "tab", { "SwitchCase": 1 } ],
    "key-spacing": [ "error", { "beforeColon": false, "afterColon": true } ],
    "keyword-spacing": "error",
    "linebreak-style": [ "error", "unix" ],
    "new-cap": [ "error", { "newIsCap": true, "capIsNew": false, "properties": true } ],
    "new-parens": "error",
    "no-alert": "error",
    "no-array-constructor": "error",
    "no-bitwise": "error",
    "no-caller": "error",
    "no-catch-shadow": "error",
    "no-constant-condition": [ "error", { "checkLoops": false } ],
    "no-empty": [ "error", { "allowEmptyCatch": true } ],
    "no-eval": "error",
    "no-extend-native": "error",
    "no-extra-bind": "error",
    "no-extra-label": "error",
    "no-floating-decimal": "error",
    "no-implicit-coercion": [ "error", { "string": true, "boolean": false, "number": false } ],
    "no-implicit-globals": "error",
    "no-implied-eval": "error",
    "no-label-var": "error",
    "no-loop-func": "error",
    "no-multi-spaces": "error",
    "no-multi-str": "error",
    "no-multiple-empty-lines": [ "error", { "max": 1, "maxBOF": 0, "maxEOF": 0 } ],
    "no-native-reassign": "error",
    "no-negated-in-lhs": "error",
    "no-new": "error",
    "no-new-func": "error",
    "no-new-object": "error",
    "no-new-require": "error",
    "no-new-wrappers": "error",
    "no-octal-escape": "error",
    "no-proto": "error",
    "no-return-assign": "error",
    "no-script-url": "error",
    "no-self-compare": "error",
    "no-sequences": "error",
    "no-shadow-restricted-names": "error",
    "no-throw-literal": "error",
    "no-trailing-spaces": "error",
    "no-undef-init": "error",
    "no-underscore-dangle": "error",
    "no-unmodified-loop-condition": "error",
    "no-unneeded-ternary": [ "error", { "defaultAssignment": false } ],
    "no-unused-expressions": "error",
    "no-use-before-define": "error",
    "no-useless-call": "error",
    "no-useless-computed-key": "error",
    "no-useless-concat": "error",
    "no-void": "error",
    "no-whitespace-before-property": "error",
    "no-with": "error",
    "object-curly-spacing": [ "error", "always" ],
    "one-var": [ "error", "always" ],
    "operator-linebreak": [ "error", "after" ],
    "prefer-numeric-literals": "error",
    "quote-props": [ "error", "as-needed", { "keywords": true } ],
    "quotes": [ "error", "single" ],
    "semi": [ "error", "always" ],
    "semi-spacing": [ "error", { "before": false, "after": true } ],
    "space-before-blocks": [ "error", "always" ],
    "space-before-function-paren": [ "error", { "anonymous": "always", "named": "never" } ],
    "space-in-parens": [ "error", "always", { "exceptions": [ "empty" ] } ],
    "space-infix-ops": "error",
    "space-unary-ops": [ "error", { "words": true, "nonwords": false } ],
    "spaced-comment": [ "error", "always", { "exceptions": [ "*", "!" ] } ],
    "unicode-bom": [ "error" ],
    "valid-jsdoc": [ "error", { "requireParamDescription": false, "requireReturnDescription": false, "requireReturn": false, "prefer": { "returns": "return" } } ],
    "vars-on-top": "error",
    "wrap-iife": "error",
    "yoda": [ "error", "never" ]
  }
}
```
</details>

**What did you do?**

```js
  // eslint-disable-next-line valid-jsdoc
  /**
   * @param {number} a
   * @param {number} b
   * @returns {undefined}
   */
  Example.sum = function ( a, b ) {
    return a + b;
  };
```

**What did you expect to happen?**

The warning for using `@returns` instead of `@return` should have been ignored.

**What actually happened?**

Got `"Use @return instead."`.

**Anything else?**

This last worked when Wikimedia used eslint `^3.1.1`, but when switching to `^4.9.0` it no longer works as expected. I actually find it an improvement that the error is now attributed to the correct line, instead of the line where the documentation comment starts. However, that also caused the `disable-next-line` logic to no longer work as expected.

I'm mainly looking for what the best practice is for a case like this, rather than making it work as-is. I'm fine with changing the comments.

Lastly, please note that the above example is not production code. I do not foresee any production use case in warning to ignore violations like this. The reason I am doing this is because of the unit tests for ESLint rules we have. See <https://github.com/wikimedia/eslint-config-wikimedia/tree/v0.5.0/test/fixtures>.

The way I have it set up, is to have a file with violations, and comments to disable those violations. And asserting that by default it passes, and that when disabling inline config, it fails with the expected violations.

**Comment by platinumazure** [comment]
Hi @Krinkle, thanks for the issue.

At this point, we don't have any comments that focus on the next "token" or "node" (or comment, in this case). So the only options are to disable and enable the rule before/after the comment, or to disable the rule for the whole file. It's probably the case that neither of these is a great experience for your use case.

Disabling before and after:

```js
/* eslint-disable valid-jsdoc */

/**
 * JSDoc comment. Enable comment could go later as well.
 */ /* eslint-enable valid-jsdoc */
function foo() {
}
```

Disabling rule for the file:

```js
// anywhere in the file:
/* eslint valid-jsdoc: off */

// ...

/**
 * JSDoc comment.
 */
function foo() {
}
```

**Comment by nzakas** [comment]
It looks like the conversation is stalled here. As this is a question rather than an action item, I'm closing the issue. If you still need help, please send a message to our [mailing list](https://groups.google.com/group/eslint) or [chatroom](https://gitter.im/eslint/eslint). Thanks!

**Comment by dashaki** [comment]
> Hi @Krinkle, thanks for the issue.
> Disabling before and after:
> 
> ```js
> /* eslint-disable valid-jsdoc */
> 
> /**
>  * JSDoc comment. Enable comment could go later as well.
>  */ /* eslint-enable valid-jsdoc */
> function foo() {
> }
> ```
This method does not work. It results in the "Missing JSDoc" error.
